x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java

```diff
@@ -39,6 +39,15 @@
                     )
                     .privileges("write", "create_index", "auto_configure")
                     .build(),
+                RoleDescriptor.IndicesPrivileges.builder()
+                    // APM Server (and hence Fleet Server, which issues its API Keys) needs additional privileges
+                    // for the non-sensitive "sampled traces" data stream:
+                    // - "maintenance" privilege to refresh indices
+                    // - "monitor" privilege to be able to query index stats for the global checkpoint
+                    // - "read" privilege to search the documents
+                    .indices("traces-apm.sampled-*")
+                    .privileges("read", "write", "monitor", "create_index", "auto_configure", "maintenance")
```

Comment:
This is my personal preference: Since `write`, `create_index` and `auto_configure` is already granted with the broader index pattern (`traces-*`) a few line above. We can remove them from this line, i.e.:
```suggestion
                    .privileges("read", "monitor", "maintenance")
```
It's not a strong opinion since the listing them again has better readability. But recent work on authorization performance led me to believe the above suggested format could have slightly better performance.

Reply:
Sounds good, thanks!